### PR TITLE
fix backup failed when pod was auto restarted by k8s

### PR DESCRIPTION
**Pull request by WizardXiao**
### What problem does this PR solve?

Ref #4805 
### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
- [ ] No code 

### k8s may restart backup pod which is not done.
1. drain node, node's pod will be schedule to other node
2. delete pod when job still exist.
### detail test
drain node test
1. backup is running at `worker2`, drain `worker2`, backup restart at `worker3` by k8s.
image
image
2. clean data before run br command to backup data
image
3. backup success
image

delete pod test
1. backup pod was deleted and then job restart a new pod
image
2. clean data before run br command to backup data in the new pod 
image
3. backup success
image
### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
fix backup failed when pod was auto restarted by k8s
```

### CODE REVIEW

File: cmd/backup-manager/app/backup/backup.go [hunk 1]
```diff
@@ -112,13 +115,39 @@ func (bo *Options) backupData(
 		go bo.updateProgressFromFile(progressCtx.Done(), backup, progressFile, progressStep, statusUpdater)
 	}
 
+	if backup.Spec.Mode == v1alpha1.BackupModeSnapshot && !bo.isBRCanContinueRunByCheckpoint() {
```
Comment (WangLe1321) [inline cmd/backup-manager/app/backup/backup.go:118]: Can we check whether the backup is restarted? I worry that we may delete valid backup data. I think we shouldn't clean backup data at first time. For example, user starts a backup with `retain` policy, then user deletes that backup. If user starts a backup with same name, we deletes valid backup data.

Comment (WizardXiao) [reply cmd/backup-manager/app/backup/backup.go:118]: ok, i think this commit can check this and only when backup is already running status, we will clean
https://github.com/pingcap/tidb-operator/pull/4883/commits/ffc46ec7178acc6fd22a7801c69ec1d2d99b6c4a

Comment (WizardXiao) [reply cmd/backup-manager/app/backup/backup.go]: i add running status check in https://github.com/pingcap/tidb-operator/pull/4883/commits/acecff3b4d6853b7f5a37e963cd53533aeaed370, PTAL

**Review by fengou1 — APPROVED** [review]

File: cmd/backup-manager/app/backup/backup.go (hunk 1 above)
Comment (WizardXiao) [reply cmd/backup-manager/app/backup/backup.go]: after a discuss, i use `!scheduled` as the judging condition and add `restart` condition as a record.

**Review by grovecai — APPROVED** [review]
LGTM, pls add e2e test

**Pull request merged**